### ISSUE TITLE
Explain data output format

### ISSUE THREAD

**Issue by akshay-joshi** [akshay-joshi/TestRMToHub#613]
**This issue has been migrated from Redmine.**

Author Name: **Adam Brusselback** (Adam Brusselback)
Original Redmine Issue: [1343](https://redmine.postgresql.org/issues/1343)

---
I cannot see a way to change the format that explain generates. Currently it's json, is there a way to change that to any of the other formats?

**Comment by akshay-joshi** [comment]
Author Name: **Sanket Mehta** (Sanket Mehta)



---

-  status_id  was changed from    New   to  In Progress

**Comment by akshay-joshi** [comment]
Author Name: **Sanket Mehta** (Sanket Mehta)


---
For graphical explain only JSON format is supported

User can see explain output in different format like TEXT, XML, YAML but for that format no graphical explain will be available

To see explain in different format use below query:

EXPLAIN (FORMAT TEXT | XML | JSON | YAML) <Query>
and press execute rather than explain button.

For more information use below link:
https://www.postgresql.org/docs/current/static/sql-explain.html

**Comment by akshay-joshi** [comment]
Author Name: **Sanket Mehta** (Sanket Mehta)



---

-  status_id  was changed from    In Progress   to  Resolved

**Comment by akshay-joshi** [comment]
Author Name: **Dave Page** (@dpage)


---
I think the point is that the "Data Output" tab which shows the raw explain output does so as a single line of JSON which is pretty much unreadable.

We shouldn't modify the graphical explain to use other formats of course, but perhaps we should think about how to improve the text display in parallel.

---

-  status_id  was changed from    Resolved   to  New

**Comment by akshay-joshi** [comment]
Author Name: **Ashesh Vashi** (@asheshv)


---
Dave,

JSON was not shown in the single line, but - the formatting was lost due to the change in the commit-id: f9393aee4488c5f35eece0d2884ccf90ff13d7bf.

In that commit, we did not want to show the footer space used by the pager, when no of records is set to 0 (zero), or less.
Hence - we were removing the class 'sql-editor-grid-container', but that was not correct solution, as there were CSS of few children DOM Element depends on that class, and it was loosing the formatting CSS.
I've fixed that issue now.


And, regarding support for other format needs to be added, but, not now.
We will be adding that support in next release (I guess).

**Comment by akshay-joshi** [comment]
Author Name: **Dave Page** (@dpage)


---
Confirmed we see multiline output now. Moving to feature tracker.

---

-  tracker_id  was changed from    Bug   to  Feature

**Comment by akshay-joshi** [comment]
Author Name: **nicolas boisteault** (nicolas boisteault)


---
Hi,
I sometimes use https://explain.depesz.com/ to find bottlenecks in my SQL.
My problem is that it doesn't take JSON as input.
I tried :
Explain (ANALYZE ON, VERBOSE OFF, COSTS OFF, BUFFERS OFF, TIMING OFF) My query...
but the results is on multiple lines which I can't copy as a whole text to paste.
It would be great to have the ability to choose the output format, maybe a warning could tell the graphical explain is disabled in this case.
Thank you for this nice tool.

**Comment by akshay-joshi** [comment]
Author Name: **Akshay Joshi** (@akshay-joshi)



---

-   Sprint  was changed    to Ready

**Comment by akshay-joshi** [comment]
Author Name: **Akshay Joshi** (@akshay-joshi)


---
We have changed the backgrid to slick grid, so now it is showing multiline output in proper JSON format.

---

-  status_id  was changed from    New   to  Rejected
